### PR TITLE
Sign In flow on Server

### PR DESCRIPTION
**Pull request by GarryFirst-cripto**
Implement Sign In flow on the server

### CODE REVIEW

File: backend/src/api/routes/authRoutes.ts [hunk 1]
```diff
@@ -1,10 +1,14 @@
...
-import { register } from '../../services/authService';
+import { login, register } from '../../services/authService';
+import authenticationMiddleware from '../middlewares/authenticationMiddleware';
 
 const router = Router();
 
 router
+  .post('/login', authenticationMiddleware, (req, res, next) => login(req.user)
+    .then(data => res.send(data))
+    .catch(next))
```
Comment (vikachuu) [inline backend/src/api/routes/authRoutes.ts:11]: You should use `run` helper method from `/common/utils/routeHelper` here.

File: backend/src/common/models/user/IRegisterUser.ts [hunk 2]
```diff
@@ -3,3 +3,7 @@ export interface IRegisterUser {
   email: string;
   password: string;
 }
+
+export interface IUserId {
+  id: string;
+}
```
Comment (vikachuu) [inline backend/src/common/models/user/IRegisterUser.ts:9]: It's better to follow the rule "one file for one interface"

File: backend/src/api/middlewares/authenticationMiddleware.ts [hunk 3]
```diff
@@ -0,0 +1,3 @@
+import passport from 'passport';
+
+export default passport.authenticate('login', { session: false });
```
Comment (AlikRakhmonov) [inline backend/src/api/middlewares/authenticationMiddleware.ts:3]: I don't see strategy for the login. Did you test the functionality?

File: backend/src/services/authService.ts [hunk 4]
```diff
@@ -1,11 +1,19 @@
 import { getCustomRepository } from 'typeorm';
 
 import UserRepository from '../data/repositories/userRepository';
-import { IRegisterUser } from '../common/models/user/IRegisterUser';
+import { IRegisterUser, IUserId } from '../common/models/user/IRegisterUser';
 import { encrypt } from '../common/utils/encryptHelper';
 import { fromUserToUserClient, fromRegisterUserToCreateUser } from '../common/mappers/user';
 import { createToken } from '../common/utils/tokenHelper';
 
+export const login = async ({ id }: IUserId) => {
```
Comment (AlikRakhmonov) [inline backend/src/services/authService.ts:9]: interface with one prop doesn't make sence. Also, where did you get ID?

**Pull request closed without merging**